python/packages/ai/teams/ai/models/openai_model.py

```diff
@@ -202,77 +202,15 @@ async def complete_prompt(
         if self._options.logger is not None:
             self._options.logger.debug(f"PROMPT:\n{res.output}")
 
-        messages: List[chat.ChatCompletionMessageParam] = []
-
-        for msg in res.output:
-            param: Union[
-                chat.ChatCompletionUserMessageParam,
-                chat.ChatCompletionAssistantMessageParam,
-                chat.ChatCompletionSystemMessageParam,
-                chat.ChatCompletionToolMessageParam,
-            ] = chat.ChatCompletionUserMessageParam(
-                role="user",
-                content=msg.content if msg.content is not None else "",
-            )
-
-            if msg.name:
-                setattr(param, "name", msg.name)
-
-            if msg.role == "assistant":
-                param = chat.ChatCompletionAssistantMessageParam(
-                    role="assistant",
-                    content=msg.content if msg.content is not None else "",
-                )
-
-                tool_call_params: List[chat.ChatCompletionMessageToolCallParam] = []
-
-                if msg.action_calls and len(msg.action_calls) > 0:
-                    for tool_call in msg.action_calls:
-                        tool_call_params.append(
-                            chat.ChatCompletionMessageToolCallParam(
-                                id=tool_call.id,
-                                function=Function(
-                                    name=tool_call.function.name,
-                                    arguments=tool_call.function.arguments,
-                                ),
-                                type=tool_call.type,
-                            )
-                        )
-                    param["content"] = None
-                    param["tool_calls"] = tool_call_params
-
-                if msg.name:
-                    param["name"] = msg.name
-
-            elif msg.role == "tool":
-                param = chat.ChatCompletionToolMessageParam(
-                    role="tool",
-                    tool_call_id=msg.action_call_id if msg.action_call_id else "",
-                    content=msg.content if msg.content else "",
-                )
-            elif msg.role == "system":
-                # o1 models do not support system messages
-                if is_o1_model:
-                    param = chat.ChatCompletionUserMessageParam(
-                        role="user",
-                        content=msg.content if msg.content is not None else "",
-                    )
-                else:
-                    param = chat.ChatCompletionSystemMessageParam(
-                        role="system",
-                        content=msg.content if msg.content is not None else "",
-                    )
-
-                if msg.name:
-                    param["name"] = msg.name
-
-            messages.append(param)
+        messages: List[chat.ChatCompletionMessageParam]
+        messages = self._map_messages(res.output, is_o1_model)
```

Comment:
could we merge these lines together?

Reply:
that would make the line too long and throw a lint error